lua/bufferline/jump_mode.lua

```diff
@@ -60,64 +65,71 @@ function M.assign_next_letter(bufnr)
     for i = 1, strwidth(name) do
       local letter = string_lower(strcharpart(name, i - 1, 1))
 
-      if M.index_by_letter[letter] ~= nil then
-        local index = M.index_by_letter[letter]
-        local status = M.letter_status[index]
+      if JumpMode.index_by_letter[letter] ~= nil then
+        local index = JumpMode.index_by_letter[letter]
+        local status = JumpMode.letter_status[index]
         if status == false then
-          M.letter_status[index] = true
+          JumpMode.letter_status[index] = true
           -- letter = m.letters[index]
-          M.buffer_by_letter[letter] = bufnr
-          M.letter_by_buffer[bufnr] = letter
+          JumpMode.buffer_letter_bimap[letter] = bufnr
+          JumpMode.buffer_letter_bimap[bufnr] = letter
           return letter
         end
       end
     end
   end
 
   -- Otherwise, assign a letter by usable order
-  for i, status in ipairs(M.letter_status) do
+  for i, status in ipairs(JumpMode.letter_status) do
     if status == false then
-      local letter = M.letters:sub(i, i)
-      M.letter_status[i] = true
-      M.buffer_by_letter[letter] = bufnr
-      M.letter_by_buffer[bufnr] = letter
+      local letter = JumpMode.letters:sub(i, i)
+      JumpMode.letter_status[i] = true
+      JumpMode.buffer_letter_bimap[letter] = bufnr
+      JumpMode.buffer_letter_bimap[bufnr] = letter
       return letter
     end
   end
 
   return nil
 end
 
-function M.unassign_letter(letter)
+function JumpMode.unassign_letter(letter)
   if letter == '' or letter == nil then
     return
   end
 
-  local index = M.index_by_letter[letter]
+  local index = JumpMode.index_by_letter[letter]
 
-  M.letter_status[index] = false
+  JumpMode.letter_status[index] = false
 
-  if M.buffer_by_letter[letter] ~= nil then
-    local bufnr = M.buffer_by_letter[letter]
-    M.buffer_by_letter[letter] = nil
-    M.letter_by_buffer[bufnr] = nil
+  if JumpMode.buffer_letter_bimap[letter] ~= nil then
+    local bufnr = JumpMode.buffer_letter_bimap[letter]
+    JumpMode.buffer_letter_bimap[letter] = nil
+    JumpMode.buffer_letter_bimap[bufnr] = nil
   end
+
+  JumpMode.reinitialize = true
 end
 
-function M.get_letter(bufnr)
-  if M.letter_by_buffer[bufnr] ~= nil then
-    return M.letter_by_buffer[bufnr]
+function JumpMode.get_letter(bufnr)
+  if JumpMode.buffer_letter_bimap[bufnr] ~= nil then
+    return JumpMode.buffer_letter_bimap[bufnr]
   end
 
-  return M.assign_next_letter(bufnr)
+  return JumpMode.assign_next_letter(bufnr)
 end
 
-function M.unassign_letter_for(bufnr)
-  M.unassign_letter(M.get_letter(bufnr))
+function JumpMode.unassign_letter_for(bufnr)
+  JumpMode.unassign_letter(JumpMode.get_letter(bufnr))
 end
 
 
-function M.activate()
+function JumpMode.activate()
+  if JumpMode.reinitialize then
```

Comment:
This is the actual fix. `reinitialize` is set to `true` when running `unassign_letter`, and then if we `activate` after closing any buffer, reassign all of the letters associated to open buffers.

Reply:
And can we run this when `unassign_letter` is called instead? It would avoid the `reinitialize` variable

Reply:
The variable allows us to queue the reinit operation. 

Since `unassign_letter` is called each buffer close (via autocmd), we would be reassigning all of the buffer letters each time the user closes a buffer. But the user might close more than one buffer before entering jump mode again, or they might not enter jump mode again at all before exiting. In these cases, we avoid reassigning the buffer letters until they are needed (see [here](https://github.com/romgrk/barbar.nvim/pull/257#discussion_r912371723) also).

Reply:
I don't think it would be a relevant performance cost, but ok. Let's keep it as a local variable though, we don't need to expose it.